src/components/users/AddNewUser.js

```diff
@@ -0,0 +1,136 @@
+import axios from "axios";
+import { Fragment, useState } from "react";
+
+const AddNewUser = ({ users, setUsers }) => {
+  const [name, setName] = useState("");
```

Comment:
too many states. create a user object and then put all the states there as keys and values.